### ISSUE TITLE
Integration tests failing on master

### ISSUE THREAD

**Issue by greenape** [Flowminder/FlowKit#451]
e.g. https://circleci.com/gh/Flowminder/FlowKit/27969

```
assert False  +  where False = cache_schema_is_empty(<flowmachine.core.connection.Connection object at 0x7f44c0a840b8>)
zmq_url = 'tcp://localhost:5555'
fm_conn = <flowmachine.core.connection.Connection object at 0x7f44c0a840b8>
redis = Redis<ConnectionPool<Connection<host=localhost,port=6379,db=0>>>

    @pytest.mark.asyncio
    async def test_run_query(zmq_url, fm_conn, redis):
        """
        Run daily_location query and check the resulting table contains the expected rows.
        """
        msg_run_query = {
            "action": "run_query",
            "query_kind": "daily_location",
            "params": {
                "date": "2016-01-01",
                "daily_location_method": "last",
                "aggregation_unit": "admin3",
                "subscriber_subset": "all",
            },
            "request_id": "DUMMY_ID",
        }
        expected_query_id = "e39b0d45bc6b46b7700c67cd52f00455"
    
        #
        # Check that we are starting with a clean slate (no cache tables, empty redis).
        #
>       assert cache_schema_is_empty(fm_conn)
E       assert False
E        +  where False = cache_schema_is_empty(<flowmachine.core.connection.Connection object at 0x7f44c0a840b8>)

tests/flowmachine_server_tests/test_action_run_query.py:50: AssertionError
```

Cache schema is showing as not empty when it should be.

**Comment by greenape** [comment]
The check is using sqlalchemy's inspector, which does some caching stuff and hence isn't super reliable. But I don't think it can be that, because the cache seems to be per `Inspector` object and you get a new inspector for each call to `inspect(engine)`.

**Comment by maxalbert** [comment]
Is this a reproducible error? I remember seeing this before, but only in (relatively) recent CI runs. I wonder whether some recent changes to the caching internals have led to the `cache_schema_is_empty()` function being out of sync with how the internals work, but if it's not a reproducible error then it's obviously difficult to bisect the history to see when this first started showing up.

**Comment by greenape** [comment]
Could also be an ordering issue?

@maxalbert it comes and goes at random. Very irritating.

**Comment by maxalbert** [comment]
This was fixed in 836b641f8c00122f53f1d80f5e3c04ec4829707c (the intermittent failures were due to the fact that sqlalchemy didn't always return the table names in the same order).
